### PR TITLE
Custom Header 🤖

### PR DESCRIPTION
**Pull request by moralesbang**
### Summary

Add custom header for Library's stack navigator.

### Screenshots

Screen Shot 2020-05-27 at 6 49 22 PM

### Trello Card

https://trello.com/c/CFxVzEM3

### CODE REVIEW

File: src/app/index.tsx [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
 
-import BookList from './screens/BookList';
 import BookDetail from './screens/BookDetail';
+import BookList from './screens/BookList';
+import styles from './styles';
 
 const { Navigator, Screen } = createStackNavigator<RootStackParamList>();
 
 function App() {
+  const stackNavigatorConfig = {
```
Comment (marianozicavo) [inline src/app/index.tsx:18]: Let's move this to `src/config/navigation.tsx`. You can put the styles in that same file. Like 
```
stackNavigatorConfig:{
 headerTitleStyle: {
     fontSize: 16,
     color: white,
     fontWeight: '600',
     letterSpacing: -0.4
 },
...
```

File: src/app/index.tsx [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
+    headerTitleStyle: styles.headerTitle,
+    headerStyle: styles.header,
+    headerBackTitleVisible: false,
+    headerBackground: () => <Image source={headerBackground} style={styles.headerBackground} />,
```
Comment (marianozicavo) [inline src/app/index.tsx:22]: Create a component for this in @components

File: src/app/index.tsx [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
+    headerBackImage: () => <Image source={goBackIcon} style={styles.goBackButton} />
```
Comment (marianozicavo) [inline src/app/index.tsx:23]: Create a component for this in @components

File: src/app/index.tsx (hunk 1 above)
Comment (moralesbang) [reply src/app/index.tsx:18]: I like that

File: package.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
     "test": "jest",
     "lint": "eslint . --ext .js,.ts,.jsx,.tsx",
     "lint-fix": "eslint . --fix --ext .js,.jsx,.ts,.tsx",
-    "lint-diff": "git diff --name-only --cached --relative | grep -E \"\\.(ts|tsx|js|jsx)$\" | xargs eslint",
+    "lint-diff": "git diff --name-only --cached --no-error-on-unmatched-pattern --relative | grep -E \"\\.(ts|tsx|js|jsx)$\" | xargs eslint",
```
Comment (sfernandez11) [inline package.json:14]: 👍

File: src/app/components/BackButton/styles.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { StyleSheet } from 'react-native';
+
+export default StyleSheet.create({
+  backButton: {
+    width: 20,
+    height: 20,
```
Comment (sfernandez11) [inline src/app/components/BackButton/styles.ts:6]: Extract common  value

**Pull request merged**